File: 1463-the-k-weakest-rows-in-a-matrix/1463-the-k-weakest-rows-in-a-matrix.java
class Solution {
    public int[] kWeakestRows(int[][] mat, int k) {
        PriorityQueue<int[]> pq = new PriorityQueue<>(
            (a, b) -> a[0] != b[0] ? Integer.compare(a[0], b[0]) : Integer.compare(a[1], b[1])
        );
        
        int n = mat.length;
        int m = mat[0].length;
        
        for (int i = 0; i < n; i++) {
            int soldiers = 0;
            for (int j = 0; j < m; j++) {
                if (mat[i][j] == 1) soldiers++;
                else break;
            }
            pq.add(new int[]{soldiers, i});
        }

        int[] weakest = new int[k];
        int index = 0;
        while (k-- > 0) {
            int[] r = pq.poll();
            weakest[index++] = r[1];
        }
        return weakest;
    }
}

File: 1463-the-k-weakest-rows-in-a-matrix/1463-the-k-weakest-rows-in-a-matrix.cpp
class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        priority_queue<pair<int,int>,vector<pair<int,int>>,greater<pair<int,int>>> pq;
        
        int n = mat.size();
        int m = mat[0].size();
        
        for(int i=0;i<n; i++){
            int soldiers=0;
            for(int j=0;j<m; j++){
                if(mat[i][j]==1) soldiers++;
            }
            pq.push({soldiers, i});
        }

        vector<int> weakest;
        while(k--){
            auto r = pq.top(); pq.pop();
            weakest.push_back(r.second);
        }
        return weakest;
    }
};